authn/saml/src/main/java/se/swedenconnect/signservice/authn/saml/SwedenConnectSamlAuthenticationHandler.java

```diff
@@ -225,6 +272,54 @@ protected boolean isSignMessageSupported() {
     return true;
   }
 
+  /**
+   * Checks if the IdP declares the entity category "http://id.elegnamnden.se/sprop/1.0/scal2".
+   */
+  @Override
+  protected boolean isSignatureActivationProtocolSupported(@Nonnull final EntityDescriptor idpMetadata) {
+
+    return EntityDescriptorUtils.getEntityCategories(idpMetadata).stream()
+        .filter(e -> EntityCategoryConstants.SERVICE_PROPERTY_CATEGORY_SCAL2.getUri().equals(e))
+        .findFirst()
+        .isPresent();
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  protected void resetContext(@Nonnull final SignServiceContext context) {
+    super.resetContext(context);
+
+    context.remove(SAD_ID_KEY);
+  }
+
+  /**
+   * Checks for a SAD request.
+   */
+  @Override
+  protected void extendedAssertionVerification(@Nonnull final AuthnRequirements authnRequirements,
+      @Nonnull final AuthnRequest authnRequest, @Nonnull ResponseProcessingResult result,
+      @Nonnull final SignServiceContext context) throws UserAuthenticationException {
+
+    // Did we include a SADRequest extension in the AuthnRequest?
+    //
+    final String sadId = context.get(SAD_ID_KEY, String.class);
+    if (sadId == null) {
+      return;
+    }
+
```

Comment:
Shouldn't the method be "extendedSADVerification" ? 
If this is a generic function then perhaps we should not return immediately if sadId == null
What if there are more checks to be done?

Reply:
But that's the only thing we do in this implementation. The method is defined in the abstract base class, and that class knows nothing about SAD. Another implementation may wish to extend this.

In other words. It is correct.